redfish-core/lib/fabric_adapters.hpp

```diff
@@ -173,11 +173,37 @@ inline void
     });
 }
 
+inline void
+    linkAsPCIeDevice(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                     const std::string& fabricAdapterPath)
+{
+    const std::string pcieDeviceName =
+        sdbusplus::message::object_path(fabricAdapterPath).filename();
+
+    if (pcieDeviceName.empty())
+    {
+        BMCWEB_LOG_ERROR("Failed to find / in pcie device path");
+        messages::internalError(asyncResp->res);
+        return;
+    }
+
+    nlohmann::json::array_t deviceArray;
+    nlohmann::json& device = deviceArray.emplace_back();
+
+    device["@odata.id"] = boost::urls::format(
+        "/redfish/v1/Systems/system/PCIeDevices/{}", pcieDeviceName);
```

Comment:
I think the current code is first add an empty item and then update it.
Instead, I think the better approach may be:

```
  nlohmann::json::object_t device;

    device["@odata.id"] = boost::urls::format(
        "/redfish/v1/Systems/system/PCIeDevices/{}", pcieDeviceName);

   deviceArray.emplace_back( device );
```